### PR TITLE
Feature/brower list as rule configuration

### PR DESCRIPTION
**Pull request by terenc3**
This adds the possibility to specify browserlist as rule configuration in eslint e.g.
```js
{
  // ...
  "rules": [
    "compat/compat": ["error", "defaults, not ie < 9"],
  ]
}
```

This is usefull if your build script is in a seperate package and you want to centralize the configuration.

Any feedback is appreciated.

### CODE REVIEW

**Comment by niksy** [comment]
@terenc3 maybe make it as object option so this can be extended with other options in future? E.g. 

```json
{
  "browsers": ["defaults", "not ie < 9"] 
}
```

**Comment by terenc3** [comment]
@niksy afaik eslint only support the array notation for options, see https://eslint.org/docs/user-guide/configuring#configuring-rules

**Comment by niksy** [comment]
@terenc3 [`switch-colon-spacing`](https://eslint.org/docs/rules/switch-colon-spacing) supports object

**Comment by terenc3** [comment]
@niksy no i will not make it and option object, because it would probaly never used. This is a single rule eslint plugin and also browserslist configuration is a "one liner". So lets keep it simple

File: src/Versioning.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ export default function DetermineTargetsFromConfig(config?: BrowserListConfig):
     return browserslist(config);
   }
 
+  if (config && typeof config === 'string') {
+    return browserslist(config.split(',').map(item => item.trim()));
```
Comment (vikr01) [inline src/Versioning.js]: Doesn't `browserslist` already support a string input? Is there a need to split the string?

Comment (terenc3) [reply src/Versioning.js]: Yes you are right, according to https://www.npmjs.com/package/browserslist#js-api it can be string or array

**Pull request closed without merging**